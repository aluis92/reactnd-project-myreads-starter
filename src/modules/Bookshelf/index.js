import React from 'react';
import {
    arrayOf,
    shape,
    string,
} from 'prop-types';
import Book from '../../elements/Book';

import {
    Container,
    BookshelfTitle,
    BookshelfBooks,
    BooksGrid,
} from './styles';

function Bookshelf({
    books,
    title,
}) {
    return (
        <Container>
            <BookshelfTitle>{title}</BookshelfTitle>
            <BookshelfBooks>
                <BooksGrid>
                    {
                        books.map(book => (
                            <Book
                                key={book.id}
                                {...book}
                            />
                        ))
                    }

                </BooksGrid>
            </BookshelfBooks>
        </Container>
    );
}

Bookshelf.defaultProps = {
    books: [],
    title: '',
};

Bookshelf.propTypes = {
    books: arrayOf(
        shape(Book.propTypes),
    ),
    title: string,
};

export default Bookshelf;
