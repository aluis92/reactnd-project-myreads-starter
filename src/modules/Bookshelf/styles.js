import styled from 'styled-components';
import arrowDropDown from '../../assets/icons/arrow-drop-down.svg'

export const Container = styled.div`
    padding: 0 10px 20px;

    @media (min-width: 600px) {
        .bookshelf {
            padding: 0 20px 40px;
        }
    }
`;

export const BookshelfTitle = styled.h2`
    border-bottom: 1px solid #dedede;
`;

export const BookshelfBooks = styled.div`
    text-align: center;
`;

export const BooksGrid = styled.ol`
    list-style-type: none;
    padding: 0;
    margin: 0;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;

    li {
        padding: 10px 15px;
        text-align: left;
    }
`;
