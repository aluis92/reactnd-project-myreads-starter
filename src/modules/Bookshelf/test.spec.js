import React from 'react';
import {
    shallow,
} from 'enzyme';

import {
    BookshelfTitle,
} from './styles';

import Bookshelf from './index';

describe('Bookshelf component', () => {
    const defaultBookshelfProps = {
        books: [],
        onChangeShelf: jest.fn(),
        title: '',
    };

    it('should render without crashing', () => {
        expect(shallow(
            <Bookshelf {...defaultBookshelfProps} />,
        )).toMatchSnapshot();
    });

    it('should books component if there is one or more books in array', () => {
        const bookshelfProps = {
            onChangeShelf: jest.fn(),
            books: [
                {
                    id: '1',
                },
                {
                    id: '2',
                },
            ],
        };

        expect(shallow(
            <Bookshelf {...bookshelfProps} />,
        ).find('Book').length).toBe(2);
    });

    it('should render shelf title if it has title prop', () => {
        const bookshelfProps = {
            title: 'shelf-title',
            onChangeShelf: jest.fn(),
        };

        expect(
            shallow(
                <Bookshelf {...bookshelfProps} />,
            ).find(BookshelfTitle).dive().find('h2')
                .text(),
        ).toEqual('shelf-title');
    });
});
