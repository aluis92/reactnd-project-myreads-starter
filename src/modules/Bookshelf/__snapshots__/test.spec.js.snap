// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookshelf component should render without crashing 1`] = `
<styled.div>
  <styled.div>
    <styled.ol />
  </styled.div>
</styled.div>
`;
